Sources/Views/MessageLabel.swift

```diff
@@ -322,6 +353,36 @@ open class MessageLabel: UILabel {
         return results
     }
 
+    /**
+     Simply check if the detector type is a .custom
+     All .custom contains an `NSRegularExpression`
+
+     - Parameters: detector: `DetectorType` that you want to execute
+
+     - Returns: Bool true if the detector type is a .custom, otherwise false
+     */
+    private func isCustomDetector(for detector: DetectorType) -> Bool {
+        if case .custom = detector { return true }
+        return false
+    }
+
+    /**
+     Take a custom detector and apply the matching to the text
+     
+     - Parameters: detector: `DetectorType` that you want to execute
+     - Parameters: text: where we apply the regular expression
+     
+     - Returns: an array of `NSTextCheckingResult` that contains all maching for this detector or nil if it fails
+     */
+    private func parse(custom detector: DetectorType, for text: NSAttributedString, in range: NSRange) -> [NSTextCheckingResult]? {
```

Comment:
`parseForMatches(with detector: DetectorType, in text: NSAttributedString, for range: NSRange) -> [NSTextCheckingResult]`